platform/MCU/rtl872x/inc/platform_config.h

```diff
@@ -47,7 +47,7 @@
 
 #define SYSTICK_IRQ_PRIORITY                7      //CORTEX_M33 Systick Interrupt
 
-#if (PLATFORM_ID == PLATFORM_P2 || PLATFORM_ID == PLATFORM_TRACKERM)
+#if HAL_PLATFORM_RTL872X
```

Comment:
This breaks compilation since it requires `hal_platform_config.h` to be previously included. What good is this check? If we are including `platform/MCU/rtl872x/inc/platform_config.h` it's probably reasonable to assume the platform is a RTL872x, and there is no variation in flash size for the platforms.